src/master/flags.hpp

```diff
@@ -81,6 +81,8 @@ class Flags : public virtual logging::Flags
   Option<std::string> modulesDir;
   std::string authenticators;
   std::string allocator;
+  float hierarchical_recovery_factor;
```

Comment:
I would suggest leaving `double`, as Mesos code generally uses `double` and `double` is slightly more difficult to use incorrectly.

Note that `0.80` in the line `constexpr float DEFAULT_AGENT_RECOVERY_FACTOR = 0.80;` is a **double** literal, not a float.